lib/cartopy/crs.py

```diff
@@ -3219,6 +3219,40 @@ def x_limits(self):
     def y_limits(self):
         return self._y_limits
 
+class Spilhaus(Projection):
+    """
+    Spilhaus World Ocean Map in a Square
```

Comment:
I think we should add a link to the story about this projection which explains some of the ideas here too and is a nice historical reference.
https://storymaps.arcgis.com/stories/756bcae18d304a1eac140f19f4d5cb3d

In particular it explains why those two points you chose below are special. Maybe even put those antipodal points int he docstring here too?